src/error.rs

```diff
@@ -0,0 +1,50 @@
+use diesel::r2d2;
+use thiserror::Error;
```

Comment:
this is an absolutely cancer way to use thiserror. Either use thiserror attributes properly and embed & create conversions from other error types properly, or use the anyhow if you don't care about underlying error content